elytron/ELY-1778-enhanced-audit-logging-feature-parity-with-legacy.adoc

```diff
@@ -0,0 +1,125 @@
+= ELY- Enhanced Audit Logging - Feature Parity with Legacy Audit Logging
+:author:            Justin Michael Cook
+:email:             jucook@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+Currently, Audit Logging in Elytron does not have the same capabilities as Legacy Security. This task is to enhance Elytron Audit Logging to have the same capabilities as Legacy Security Audit Logging through the addition of an @Audit annotation and additional audit events.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-1778[ELY-1778]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1230[EAP7-1230]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kašík]
+
+=== Testing By
+
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* There should be the ability to specify a Java class should be logged by using an @Audit Java annotation, similarly to what was https://developer.jboss.org/wiki/PicketBoxSecurityAnnotations#jive_content_id_SecurityAudit_Annotation[available in Picketbox]
+** This annotation would be available to be added onto a class for signifying when a method from that class is called it should be logged
```

Comment:
I am inspecting the legacy equivalent now and it seems that it had some parameters (array of `@Module`) which also had parameters - array of `@ModuleOption`. Though I failed to understand what was the purpose of these attributes. Can you please help me here and also edit the AD if there will be any parameters?

Reply:
From my understanding, the ```@Module``` (and ```@ModuleOption```) were to provide the class that handles the audit logging, but I think this is not needed now since the class specified by ```@Audit``` would just use the configured endpoint that would be specified when logging is enabled